scripts/trade0x-kovan.js

```diff
@@ -0,0 +1,275 @@
+  Redeem: 8,
+  Call: 9,
+})
+
+const Permit = [
+  {name: 'owner', type: 'address'},
+  {name: 'spender', type: 'address'},
+  {name: 'value', type: 'uint256'},
+  {name: 'nonce', type: 'uint256'},
+  {name: 'deadline', type: 'uint256'},
+]
+
+const expiryTime = new BigNumber(60 * 60 * 24)            // 1 day
+const usdcDecimals = 6
+const wethDecimals = 18
+
+const cmd = yargs
+  .option("makerPrivateKey", { describe: "0x Maker Private Key, wallet account[1]", type: "string", demandOption: true })
+  .argv;
+
+// create signer etherJS Object
+const makerEtherJS = new ethers.Wallet(cmd.makerPrivateKey)
+
+async function runExport() {
+  console.log("🔥🔥🔥 STARTING 0x TRADING SCRIPT 🔥🔥🔥")
+  console.log("0x order signer address: 🖋️ ", makerEtherJS.address)
+
+  let trade0xCallee = await Trade0x.at(trade0xAddress)
+  let otokenFactory = await OtokenFactory.at(otokenFactoryAddress)
+  let payableProxyController = await PayableProxyController.at(payableProxyAddress);
+  let controllerProxy = await Controller.at(controllerProxyAddress)
+  let marginPool = await MarginPool.at(marginPoolAddress);
+  let usdc = await ERC20.at(usdcAddress);
+  let weth = await WETH9.at(wethAddress);
+
+  const account = await web3.eth.getAccounts()
+  const taker = account[0]
+  const maker = account[1]
+
+  console.log("Taker address: ", taker)
+
+  if (makerEtherJS.address != maker) {
+    console.log("0x Maker address does not correspond to private key 🔴")
+    return;
+  }
+
+  const latestBlock = new BigNumber((await web3.eth.getBlock('latest')).timestamp);
+  let expiry = new BigNumber(latestBlock).plus(expiryTime)
+  let expiryDate = new Date(expiry.toNumber() * 1000)
+  expiryDate.setUTCHours(8)
+  expiryDate.setUTCMinutes(0)
+  expiryDate.setUTCSeconds(0)
+  expiry = new BigNumber(Math.floor(expiryDate.getTime() / 1000))
+
+  let callOption1;
+  let otokenAddress = await otokenFactory.getOtoken(weth.address, usdc.address, weth.address, createTokenAmount(640), expiry, false);
+  
+  if(otokenAddress == ZERO_ADDR) {
+    console.log("Otoken not found, creating one now! ⚡")
+    // deploy call option
+    await otokenFactory.createOtoken(weth.address, usdc.address, weth.address, createTokenAmount(640), expiry, false)
+    callOption1 = await Otoken.at(
+      await otokenFactory.getOtoken(weth.address, usdc.address, weth.address, createTokenAmount(640), expiry, false)
+    )
+  }
+  else {
+    callOption1 = await Otoken.at(otokenAddress)
+  }
+
+  console.log(await callOption1.name() + " " + callOption1.address)
+
+  const vaultCounter = new BigNumber((await controllerProxy.getAccountVaultCounter(taker))).plus(1)
+  const optionsToMint = createTokenAmount(1, 8)
+  const collateralToDeposit = createTokenAmount(1, wethDecimals)
+
+  // create 0x order
+  const order = createOrder(
+    EXCHANGE_ADDR,
+    makerEtherJS.address,
+    usdc.address,
+    callOption1.address,
+    new BigNumber(createTokenAmount(1, 6)),
+    new BigNumber(optionsToMint),
+    42
```

Comment:
better make this 42 a constant along with all the address `consts`, so might be easier to reuse this script in the future.